Common/src/Pollster/TaskHandler.cs

```diff
@@ -408,14 +408,14 @@ public async Task PreProcessing(CancellationToken cancellationToken)
   /// <summary>
   ///   Execution of the acquired task on the worker
   /// </summary>
-  /// <param name="agent">Agent that will be used to keep track of the requests of the worker</param>
-  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <param name="cancellationToken">Token used to cancel the method before the task is started</param>
+  /// <param name="connectionCancellationToken">Token used to cancel the connection to the worker</param>
   /// <returns>
   ///   Task representing the asynchronous execution of the method
   /// </returns>
   /// <exception cref="NullReferenceException">wrong order of execution</exception>
   /// <exception cref="ArmoniKException">worker pipe is not initialized</exception>
-  public async Task ExecuteTask(CancellationToken cancellationToken)
+  public async Task ExecuteTask(CancellationToken cancellationToken, CancellationToken connectionCancellationToken)
```

Comment:
Why not have the grace delay defined here instead of passing 2 different cancellationTokens ?

Reply:
I would set the two cancellationtokens (from the message and from sigterm) directly into the constructor of this class. That way you can have all the logic into one sigle class.

Reply:
In here, I am only using the combined one and the one to manage the grace delay